hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmMultipartKeyInfo.java

```diff
@@ -58,18 +60,19 @@ public class OmMultipartKeyInfo extends WithObjectID {
    * ------------------------------------------|
    */
   private long parentID;
+  public static final PartKeyInfo EMPTY_PART_KEY_INFO = null;
 
   /**
    * Construct OmMultipartKeyInfo object which holds multipart upload
    * information for a key.
    */
   public OmMultipartKeyInfo(String id, long creationTime,
       ReplicationConfig replicationConfig,
-      Map<Integer, PartKeyInfo> list, long objectID, long updateID) {
+      IntSparseMap<PartKeyInfo> list, long objectID, long updateID) {
     this.uploadID = id;
     this.creationTime = creationTime;
     this.replicationConfig = replicationConfig;
-    this.partKeyInfoList = new TreeMap<>(list);
+    this.partKeyInfoList = list;
```

Comment:
> In the change here, you have avoided copying a TreeMap in a OmMultipartKeyInfo constructor.  ...

@xichen01 , I mean this line.

Reply:
But here it will also copy `TreeMap`
https://github.com/apache/ozone/blob/52db80f920dcb11eb231e1cc98c909f3b0638b25/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmMultipartKeyInfo.java#L257-L264
 when `get` form RocksDB
https://github.com/apache/ozone/blob/cffa3864b974499078fbdb37506ed992319b09c5/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/multipart/S3MultipartUploadCommitPartRequest.java#L149-L150

Reply:
> when get form RocksDB

If it cannot be avoided easily, let's keep it for the moment.  Just remove the one in the constructor.